_posts/Durr_Hoyer_Post_1.md

```diff
@@ -0,0 +1,126 @@
+
+
+### The Algorithm
+A brief introduction to the Durr and Hoyer Algorithm can be derived from the Quantum Minimum Searching Algorithm outlined in 'A quantum algorithm for finding the minimum' [(1)]:
+
+Durr and Hoyer propose a computationally efficient routine to find the minimum of a table of distinct unsorted integers. The algorithm is as follows:
+
+
+![QMSA](/assets/images//DurrHoyer-QMSA.JPG "QMSA")
+
+
+#### Psuedocode
+
+```
+    t = table.flatten()
+    N = len(t)
+    y = randomuniform[1..N] #N is a integer
+    time_limit = (22.5*sqrt(N)=1.4*log(N)^2)
+    q = preprocess.simulate(t,y)
+
+    while time.clock() < time_limit:
+        for i in range(N):
+            while t[i] < t[y]:
+                i = N-i
+                if N is even
+                    q=Algorithm with Hadamard gates[0...j]
+                    y'=q[0]
+                else
+                    t=Algorithm with QFT instead of H[0...j]
+                    y'=q[0]
+        if t[y'] < t[y]:
+            y = y'
+            return
+    return y
+```
+------------------
+
+It is easy to understand the conceptual implementation in python. It is important to note steps:
+### (a,b)
+
+------------------
+
+![a](/assets/images//a.JPG)   
+
+Part a is probably the most interesting part of this algorithm from my perspective. After a brief chat with Dr. Hoyer I realized this aspect of the algorithm is a field of research in itself. The creation of qubits from a table of unique integers is an interesting question.
+
+For now, the more prevalent question is what sort of initialization we can use to verify the integrity and efficiency of this algorithm. My most immediate thought is to entangle all the qubits and superimpose them. The classical value of the integer is not represented here, only the index of the integer is captured at a quantum level on the table assuming an input datatype of BigEndian. If the goal is to test the algorithm's probabilistic integrity this plan of action will suffice.
+
+The register can be initiated as follows:
+
+![latex_imp](/assets/images//latex_information.JPG)
+
+Obviously this is unachievable on current quantum computer architecture. We only have a qubit, the above example implies we have harnessed the tritbit and so on till n bit. In reality we will register all integers as unique binary strings which will equate to their qubits.
+ 
+Luckily PrepareUniformSuperposition() in Q# does this for us easily. I have implemented it as follows:
+
+![intialize](/assets/images//intialize.JPG)
```

Comment:
I would strongly suggest looking at the [Q# style guide](https://docs.microsoft.com/en-us/quantum/contributing/style-guide?tabs=guidance) and making this an actual fenced code block in the text not a screen grab.